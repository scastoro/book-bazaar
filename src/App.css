@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap");
/* Universal Styles */
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
a {
  text-decoration: none;
  color: inherit;
}
html {
  font-size: 62.5%;
}
body {
  font-family: "Roboto", sans-serif;
  font-size: 1.4rem;
}
.btn {
  background-color: #000;
  border: none;
  font-weight: 600;
  color: #fff;
  border-radius: 1.5rem;
  cursor: pointer;
  padding: 1rem 2rem;
  cursor: pointer;
}
.btn:hover {
  opacity: 0.75;
}
/* Header styling */
header {
  color: #fff;
  width: 100%;
  padding: 2rem 5rem;
  position: absolute;
}
.navbar-container {
  display: flex;
}
.nav {
  margin-left: auto;
}
.nav a {
  margin-left: 5rem;
  font-size: 2rem;
  padding-bottom: 0.5rem;
  font-weight: 700;
}
.nav a:hover {
  border-bottom: 0.2rem solid #fff;
}
/* Main styling */
main {
  color: #fff;
  overflow: visible;
  background: linear-gradient(0deg, rgba(0, 0, 0, 0.66), rgba(0, 0, 0, 0.66)),
    url("https://images.unsplash.com/photo-1507842217343-583bb7270b66?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2190&q=80");
}
.home-container {
  height: 800px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.main-title {
  font-size: 5.5rem;
  font-weight: 900;
  width: 60rem;
  margin-bottom: 7rem;
}
.main-btn {
  cursor: pointer;
  background: rgba(3, 3, 3, 1);
  box-shadow: 0px 0px 7px rgba(255, 255, 255, 0.48);
  font-size: 2rem;
  font-weight: 700;
  border-radius: 2rem;
  color: #fff;
  border: none;
  padding: 2rem 2.5rem;
}
.main-btn:hover {
  background-color: #3d3b3b;
}
/* More info section styling */
.more-info {
  color: #030303;
  height: 500px;
  display: flex;
  background-color: #e3e1dc;
  padding: 7rem 5rem;
  justify-content: center;
  align-items: center;
}
.books {
  width: 40%;
  border-radius: 1rem;
  margin-right: 4rem;
}
.more-info-heading {
  font-size: 4rem;
  margin-bottom: 2rem;
}
.more-info-para {
  font-size: 2.3rem;
  width: 65%;
  font-weight: 600;
}
/* Shop Styling */
.shop {
  background-color: #e3e1dc;
  height: 1350px;
  padding-top: 7rem;
}
.book-container {
  margin-top: 2rem;
  padding: 2rem 6rem;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: 1fr 1fr;
  grid-row-gap: 9rem;
  grid-column-gap: 5rem;
  justify-items: center;
}
.card {
  width: 90%;
  background-color: #fff;
  border-radius: 1.5rem;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  padding: 3rem 0;
}
.card-image {
  width: 50%;
  margin-bottom: 1rem;
}
.card-heading {
  font-size: 1.8rem;
  margin-bottom: 0.6rem;
}
.card-label {
  margin-right: 1rem;
  margin-bottom: 0.6rem;
}
.card-input {
  width: 3rem;
  margin-left: 1rem;
}
.card-price {
  font-size: 1.8rem;
  font-weight: 600;
  margin-bottom: 1rem;
}
.card-btn {
  background-color: #000;
  border: none;
  font-weight: 600;
  color: #fff;
  border-radius: 1.5rem;
  cursor: pointer;
  padding: 1rem 2rem;
}
.card-btn:hover {
  opacity: 0.75;
}
.cart-status {
  background-color: #fff;
  width: 15rem;
  position: sticky;
  top: 1px;
  left: 100%;
  border-radius: 5px;
  border: 1px lightgray solid;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 1rem;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}
.cart-total {
  margin-bottom: 0.75rem;
}
.show-cart {
  padding: 0.5rem 1.5rem;
}
/* Cart styling */
.cart {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #fff;
  position: fixed;
  top: 0;
  right: 0;
  width: 40rem;
  padding: 2rem;
  border: 1px solid #c4c4c4;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 10px;
  min-height: 30rem;
}
.cart-title {
  font-size: 2rem;
}
.cart-items {
  padding-bottom: 2rem;
  margin-bottom: 1rem;
}
.item-list {
  display: grid;
  margin-top: 1rem;
  grid-template-columns: 1fr 2fr;
  grid-template-rows: repeat(3, 1fr);
  align-items: center;
  justify-items: start;
  grid-column-gap: 2rem;
}
.item-img {
  width: 85%;
  justify-self: center;
  order: -1;
  grid-row: 1 / -1;
}
.item-title {
  align-self: flex-end;
}
.item-sub-btn,
.item-add-btn {
  width: 1.5rem;
  margin-right: 0.5rem;
}
.item-input {
  width: 3rem;
  margin-right: 0.5rem;
}
.product-total {
  align-self: start;
}
.h-rule {
  margin: auto 0 1rem;
  width: 85%;
  border: 1px solid #c4c4c4;
}
.total {
  font-weight: 700;
  font-size: 1.8rem;
  margin-bottom: 1rem;
}
.cart-btns {
  width: 70%;
  display: flex;
  justify-content: space-around;
}
/* Footer styling */
footer {
  height: 6rem;
  background-color: #030303;
  color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.75rem;
}
.footer-link {
  border-bottom: 1px solid #fff;
}
/* Cart CSSTransitionGroup Styling */
.cart-enter {
  opacity: 0.01;
  transform: translateX(100%);
}
.cart-enter.cart-enter-active {
  opacity: 1;
  transform: translateX(0);
  transition: opacity 500ms ease-in, transform 500ms ease-in;
}
.cart-leave {
  opacity: 1;
  transform: translateX(0);
}
.cart-leave.cart-leave-active {
  opacity: 0.01;
  transform: translateX(100%);
  transition: opacity 300ms ease-in, transform 300ms ease-in;
}
