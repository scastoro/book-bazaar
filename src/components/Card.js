import React from "react";

const Card = ({
  title,
  img,
  price,
  count,
  id,
  onChangeHandler,
  addCartBtnHandler,
}) => {
  return (
    <section key={id} className="card">
      <h3 className="card-heading">{title}</h3>
      <img className="card image" src={img} alt={title} />
      <p className="card-price">{price}</p>
      <label className="card-label" htmlFor="quantity">
        Quantity:
      </label>
      <input
        type="number"
        id="quantity"
        className="card-input"
        value={count}
        onChange={(e) => onChangeHandler(e.target.value, id)}
      />
      <button onClick={() => addCartBtnHandler(id)}>Add to cart</button>
    </section>
  );
};

export default Card;
