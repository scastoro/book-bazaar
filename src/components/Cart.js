import React from "react";

const Cart = ({
  products,
  total,
  onChangeHandler,
  addBtnHandler,
  subBtnHandler,
  toggleCartHandler,
}) => {
  const productsList = products.map((product) => (
    <section key={product.id} className="item-list">
      <h4 className="item-title">{product.title}</h4>
      <img className="item-img" src={product.img} alt="" />
      <label htmlFor="count">Quantity: </label>
      <button
        className="item-sub-btn"
        onClick={() => subBtnHandler(product.id)}
      >
        -
      </button>
      <input
        type="number"
        id="count"
        value={product.count}
        onChange={(e) => onChangeHandler(e.target.value, product.id)}
      />
      <button
        className="item-add-btn"
        onClick={() => addBtnHandler(product.id)}
      >
        +
      </button>
      <p className="product-total">
        {"$"}
        {+product.price * +product.count}
      </p>
    </section>
  ));

  return (
    <section role="dialog" className="cart">
      <h3>Shopping Cart</h3>
      <section className="cart-items">{productsList}</section>
      <p className="total">Order total: ${total}</p>
      <button className="checkout-btn">Checkout</button>
      <button className="close-btn" onClick={toggleCartHandler}>
        Close
      </button>
    </section>
  );
};

export default Cart;
