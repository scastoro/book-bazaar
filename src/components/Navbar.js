import React from "react";
import { Link, Outlet } from "react-router-dom";

const Navbar = () => {
  return (
    <>
      <header>
        <img className="logo" src="#" alt="" />
        <nav className="nav">
          <Link className="text-link" to="/">
            Home
          </Link>
          <Link className="text-link" to="/shop">
            Shop
          </Link>
        </nav>
      </header>
      <Outlet />
    </>
  );
};

export default Navbar;
