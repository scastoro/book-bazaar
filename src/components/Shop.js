import React from "react";

const Shop = (props) => {
  return <h1>Hello from shop</h1>;
};

export default Shop;
