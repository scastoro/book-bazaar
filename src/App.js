import React from 'react';
import './App.css';

function App() {
  return (
    <h1>Hi, how are you?</h1>
  );
}

export default App;
