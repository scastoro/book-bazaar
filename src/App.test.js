import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";
import Navbar from "./components/Navbar";
import Shop from "./components/Shop";
import { MemoryRouter } from "react-router-dom";
import Card from "./components/Card";
import Cart from "./components/Cart";

const cardProps = {
  title: "foo",
  img: "",
  count: 1,
  price: "100",
  id: "1",
};

const testProducts = [
  {
    title: "foo",
    img: "",
    count: 1,
    price: 100,
    id: "20971bac-1a2b-4c6f-a8b8-d56f43e09495",
  },
  {
    title: "foo",
    img: "",
    count: 1,
    price: 50,
    id: "1f6f6d97-36e2-4972-b6a1-ff2520e1d491",
  },
];

const testTotal = 200;

describe("App component", () => {
  it("should render correctly", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
describe("Navbar component", () => {
  it("should render correctly", () => {
    const view = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );
    expect(view).toMatchSnapshot();
  });
});

describe("Shop component", () => {
  it("should render correctly", () => {
    const { container } = render(<Shop />);
    expect(container).toMatchSnapshot();
  });

  it("should update cart status on add to cart btn press", () => {
    render(<Shop />);
    const addToCart = screen.getAllByRole("button", { name: "Add to cart" });
    const cartStatus = screen.getAllByRole("status");

    userEvent.click(addToCart[0]);
    expect(cartStatus[0].textContent).toBe("Items in Cart: 1");
  });

  it("should increment count input when plus btn pressed", () => {
    render(<Shop />);
    const incrementBtn = screen.getAllByRole("button", { name: "+" });
    const input = screen.getAllByRole("textbox");

    userEvent.click(incrementBtn[0]);
    expect(input[0].value).toBe("2");
  });

  it("should decrement count input when minus btn pressed", () => {
    render(<Shop />);
    const decrementBtn = screen.getAllByRole("button", { name: "-" });
    const input = screen.getAllByRole("textbox");

    userEvent.click(decrementBtn[1]);
    expect(input[1].value).toBe("0");
  });

  it("should update count input on keypress", () => {
    render(<Shop />);
    const input = screen.getAllByRole("textbox");

    userEvent.type(input[0], "{backspace}34");
    expect(input[0].value).toBe("34");
  });

  it("should display correct price when adding product to cart", () => {
    render(<Shop />);
    const addToCart = screen.getAllByRole("button", { name: "Add to cart" });
    const goToCart = screen.getByRole("button", { name: "Go to cart" });
    userEvent.click(addToCart[1]);
    userEvent.click(addToCart[1]);
    userEvent.click(goToCart);
    const cartTotal = screen.getByText("Order total", { exact: false });

    expect(cartTotal.textContent).toBe("Order total: $50");
  });
});

describe("Product Card Component", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Card
        title={cardProps.title}
        img={cardProps.img}
        count={cardProps.count}
        price={cardProps.price}
        id={cardProps.id}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should call addCartBtnHandler", () => {
    const onClickMock = jest.fn();
    render(<Card addCartBtnHandler={onClickMock} />);
    const subBtn = screen.getByRole("button", { name: "Add to cart" });
    userEvent.click(subBtn);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it("should call onChange the correct number of times", () => {
    const onChangeMock = jest.fn();
    render(<Card onChangeHandler={onChangeMock} />);
    const input = screen.getByRole("textbox");

    userEvent.type(input, "123");

    expect(onChangeMock).toHaveBeenCalledTimes(3);
  });

  it("should call onChange with the correct argument(s) on each input", function () {
    const onChangeMock = jest.fn();
    render(<Card id={cardProps.id} onChangeHandler={onChangeMock} />);
    const input = screen.getByRole("textbox");

    userEvent.type(input, "12");

    expect(onChangeMock).toHaveBeenNthCalledWith(1, "1", "1");
    expect(onChangeMock).toHaveBeenNthCalledWith(2, "12", "1");
  });
});
describe("Cart component", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Cart products={testProducts} totalPrice={testTotal} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should call addBtnHandler", () => {
    const onClickMock = jest.fn();
    render(<Cart products={testProducts} addBtnHandler={onClickMock} />);
    const addBtn = screen.getAllByRole("button", { name: "+" });
    userEvent.click(addBtn[0]);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it("should call subBtnHandler", () => {
    const onClickMock = jest.fn();
    render(<Cart products={testProducts} subBtnHandler={onClickMock} />);
    const subBtn = screen.getAllByRole("button", { name: "-" });
    userEvent.click(subBtn[0]);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it("should call onChange the correct number of times", () => {
    const onChangeMock = jest.fn();
    render(<Cart products={testProducts} onChangeHandler={onChangeMock} />);
    const input = screen.getAllByRole("spinbutton");

    userEvent.type(input[0], "123");

    expect(onChangeMock).toHaveBeenCalledTimes(3);
  });

  it("should call onChange with the correct argument(s) on each input", function () {
    const onChangeMock = jest.fn();
    render(
      <Cart
        id={cardProps.id}
        products={testProducts}
        onChangeHandler={onChangeMock}
      />
    );
    const input = screen.getAllByRole("spinbutton");

    userEvent.clear(input[0]);
    userEvent.type(input[0], "12");

    expect(onChangeMock).toHaveBeenNthCalledWith(
      1,
      "",
      "20971bac-1a2b-4c6f-a8b8-d56f43e09495"
    );
    expect(onChangeMock).toHaveBeenNthCalledWith(
      2,
      "12",
      "20971bac-1a2b-4c6f-a8b8-d56f43e09495"
    );
  });
});
