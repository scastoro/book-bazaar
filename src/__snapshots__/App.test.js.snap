// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
<div>
  <header>
    <img
      alt=""
      class="logo"
      src="#"
    />
    <nav
      class="nav"
    >
      <a
        class="text-link"
        href="/"
      >
        Home
      </a>
      <a
        class="text-link"
        href="/shop"
      >
        Shop
      </a>
    </nav>
  </header>
  <h1>
    Hello from home
  </h1>
</div>
`;

exports[`Cart component should render correctly 1`] = `
<div>
  <section
    class="cart"
  >
    <h3>
      Shopping Cart
    </h3>
    <section
      class="cart-items"
    >
      <section
        class="item-list"
      >
        <h4
          class="item-title"
        >
          foo
        </h4>
        <img
          alt=""
          class="item-img"
          src=""
        />
        <label
          for="count"
        >
          Quantity: 
        </label>
        <button
          class="item-sub-btn"
        >
          -
        </button>
        <input
          id="count"
          type="number"
          value="1"
        />
        <button
          class="item-add-btn"
        >
          +
        </button>
        <p
          class="product-total"
        >
          $
          100
        </p>
      </section>
      <section
        class="item-list"
      >
        <h4
          class="item-title"
        >
          foo
        </h4>
        <img
          alt=""
          class="item-img"
          src=""
        />
        <label
          for="count"
        >
          Quantity: 
        </label>
        <button
          class="item-sub-btn"
        >
          -
        </button>
        <input
          id="count"
          type="number"
          value="1"
        />
        <button
          class="item-add-btn"
        >
          +
        </button>
        <p
          class="product-total"
        >
          $
          50
        </p>
      </section>
    </section>
    <p
      class="total"
    >
      Order total: $
    </p>
  </section>
</div>
`;

exports[`Navbar component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <img
          alt=""
          class="logo"
          src="#"
        />
        <nav
          class="nav"
        >
          <a
            class="text-link"
            href="/"
          >
            Home
          </a>
          <a
            class="text-link"
            href="/shop"
          >
            Shop
          </a>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <img
        alt=""
        class="logo"
        src="#"
      />
      <nav
        class="nav"
      >
        <a
          class="text-link"
          href="/"
        >
          Home
        </a>
        <a
          class="text-link"
          href="/shop"
        >
          Shop
        </a>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Product Card Component should render correctly 1`] = `
<div>
  <section
    class="card"
  >
    <h3
      class="card-heading"
    >
      foo
    </h3>
    <img
      alt="foo"
      class="card image"
      src=""
    />
    <p
      class="card-price"
    >
      100
    </p>
    <label
      class="card-label"
      for="quantity"
    >
      Quantity:
    </label>
    <input
      class="card-input"
      id="quantity"
      type="number"
      value="1"
    />
    <button>
      Add to cart
    </button>
  </section>
</div>
`;

exports[`Shop component should render correctly 1`] = `
<div>
  <section
    class="shop"
  >
    <section
      class="cart"
      role="status"
    >
      Items in Cart: 
      0
    </section>
    <button
      class="show-cart"
    >
      Go to cart
    </button>
    <section
      class="book-container"
    >
      <section
        class="card"
      >
        <h3
          class="card-heading"
        >
          test book 1
        </h3>
        <img
          alt="test book 1"
          class="card image"
          src=""
        />
        <p
          class="card-price"
        >
          25
        </p>
        <label
          class="card-label"
          for="quantity"
        >
          Quantity:
        </label>
        <input
          class="card-input"
          id="quantity"
          type="number"
          value="1"
        />
        <button>
          Add to cart
        </button>
      </section>
      <section
        class="card"
      >
        <h3
          class="card-heading"
        >
          test book 2
        </h3>
        <img
          alt="test book 2"
          class="card image"
          src=""
        />
        <p
          class="card-price"
        >
          25
        </p>
        <label
          class="card-label"
          for="quantity"
        >
          Quantity:
        </label>
        <input
          class="card-input"
          id="quantity"
          type="number"
          value="1"
        />
        <button>
          Add to cart
        </button>
      </section>
      <section
        class="card"
      >
        <h3
          class="card-heading"
        >
          test book 3
        </h3>
        <img
          alt="test book 3"
          class="card image"
          src=""
        />
        <p
          class="card-price"
        >
          25
        </p>
        <label
          class="card-label"
          for="quantity"
        >
          Quantity:
        </label>
        <input
          class="card-input"
          id="quantity"
          type="number"
          value="1"
        />
        <button>
          Add to cart
        </button>
      </section>
      <section
        class="card"
      >
        <h3
          class="card-heading"
        >
          test book 4
        </h3>
        <img
          alt="test book 4"
          class="card image"
          src=""
        />
        <p
          class="card-price"
        >
          25
        </p>
        <label
          class="card-label"
          for="quantity"
        >
          Quantity:
        </label>
        <input
          class="card-input"
          id="quantity"
          type="number"
          value="1"
        />
        <button>
          Add to cart
        </button>
      </section>
      <section
        class="card"
      >
        <h3
          class="card-heading"
        >
          test book 5
        </h3>
        <img
          alt="test book 5"
          class="card image"
          src=""
        />
        <p
          class="card-price"
        >
          25
        </p>
        <label
          class="card-label"
          for="quantity"
        >
          Quantity:
        </label>
        <input
          class="card-input"
          id="quantity"
          type="number"
          value="1"
        />
        <button>
          Add to cart
        </button>
      </section>
      <section
        class="card"
      >
        <h3
          class="card-heading"
        >
          test book 6
        </h3>
        <img
          alt="test book 6"
          class="card image"
          src=""
        />
        <p
          class="card-price"
        >
          25
        </p>
        <label
          class="card-label"
          for="quantity"
        >
          Quantity:
        </label>
        <input
          class="card-input"
          id="quantity"
          type="number"
          value="1"
        />
        <button>
          Add to cart
        </button>
      </section>
    </section>
  </section>
</div>
`;
